### PR TITLE
Implement delete batch job API

### PR DESCRIPTION
**Pull request by Ladicle**

### CODE REVIEW

File: server/internal/server/batch_server.go [hunk 1]
```diff
@@ -185,6 +185,45 @@ func (s *S) GetBatchJob(ctx context.Context, req *v1.GetBatchJobRequest) (*v1.Ba
...
+		}
+		return nil, status.Errorf(codes.Internal, "get batch job: %s", err)
+	}
+	jobProto, err := job.V1BatchJob()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "convert batch job to proto: %s", err)
+	}
+
+	if job.State == store.BatchJobStateDeleted ||
+		job.State == store.BatchJobStateQueued && job.QueuedAction == store.BatchJobQueuedActionDelete {
```
Comment (kkaneda) [inline server/internal/server/batch_server.go:212]: The `if` condition has both `||` and `&&`. Need to have parenthesis?

E.g.)

```suggestion
		(job.State == store.BatchJobStateQueued && job.QueuedAction == store.BatchJobQueuedActionDelete) {
```

Or maybe we can just remove `job.State == store.BatchJobStateQueued &&`? Not sure.

File: server/internal/server/batch_server.go [hunk 2]
```diff
@@ -346,6 +387,16 @@ func (ws *WS) UpdateBatchJobState(ctx context.Context, req *v1.UpdateBatchJobSta
 		}); err != nil {
 			return nil, status.Errorf(codes.Internal, "mutate batch job: %s", err)
 		}
+	case v1.InternalBatchJob_DELETED:
+		if job.State != store.BatchJobStateQueued && job.QueuedAction != store.BatchJobQueuedActionDelete {
+			// Queued state is only available in the store object and does not exist in the proto object.
+			return nil, status.Errorf(codes.FailedPrecondition, "job state is not canceling: %s (%s)", job.State, job.QueuedAction)
```
Comment (kkaneda) [inline server/internal/server/batch_server.go:393]:
```suggestion
			return nil, status.Errorf(codes.FailedPrecondition, "job state is not queued: %s (%s)", job.State, job.QueuedAction)
```

**Review by kkaneda — APPROVED** [review]
LG!

File: server/internal/server/batch_server.go (hunk 1 above)
Comment (Ladicle) [reply server/internal/server/batch_server.go]: Parenthesis is not necessary, but yeah, it makes it easier to understand.

**Pull request merged**